storages/python/python-storage-plugins/redis.py

```diff
@@ -31,7 +34,7 @@ def open(self, uri: str, options=None):
               Generate the key with `crystallography.fernet.generate_key()`.
         """
         self.uri = uri
-        opts = Options(options, defaults="port=6379;ssl=false;db=0")
+        opts = Options(options, defaults="port=6379;socket_keepalive=true;db=0")
```

Comment:
Is the point to always maintain connection? is that a good idea?

Reply:
Not really. It is more practical. The point is that I had some issues with Redis was hanging for some reasons when I was testing it. Googling the issue suggested to set `socket_keepalive=true`. Trying it solved my problem, so I simply add it as a default.

Reply:
but wouldn't it still hang when there are concurrent connections from multiple clients?- I am thinking from a deployment perspective and wouldn't it consume more resources and increase traffic because of keep-alive packets?

Reply:
I would expect redis to be able to handle many simultaneous connections, so using keep-alive should not be a problem. I looked for the time interval for the keep-alive messages but wasn't able to find it. I expect it to be on the second or minute timescale, so that shouldn't cause traffic load problems, either.

Reply:
I agree that it is strange to have to provide `keep-alive`. But without this option, running the tests hangs on my machine. It might very well be a fix for a symptom rather than the cause. But setting `keep-alive` was the suggestion I found when googling for redis hanging - and it actually works for me.  

Would a compromise be to accept this PR for now, but add an issue to investigate the real cause of the hanging problem?

Reply:
I do not have a problem with using `keep-alive`, since it increase robustness. It can surely reduce problems that are related to the network (and its policies) between the client and the server.

The PR is not linked to an issue, and the PR description does not describe whether the problems experienced with redis occur when redis is on a remote or local computer. Can you please clarify this?

Reply:
The problem is as far as I can say not consistent. I have from time to time seen it on GitHub CI/CD. During the initial development of the plugin I had no problem with redis hanging, but then it started while I was working on unrelated things.

Reply:
> but wouldn't it still hang when there are concurrent connections from multiple clients?- I am thinking from a deployment perspective and wouldn't it consume more resources and increase traffic because of keep-alive packets?

You might very well be right that setting keep-alive might not be optimal in deployment. But this PR just change the default for robustness. It is easy to set keep-alive=false in production.

Reply:
I don't see any problems with keep-alive. Those packages will be sent infrequently and only when there is low activity on the connection. There is no need to look further into this at this point.